gsoc-2023/ideas.md

```diff
@@ -0,0 +1,263 @@
+The length of the project (175 vs. 350 hours) can be negotiated based on the number of VMMs supported and which drivers.
+
+#### Reading & Related Material
+
+ * https://dl.acm.org/doi/10.1145/3447786.3456248 (in particular Figure 10)
+ * https://github.com/cloud-hypervisor/cloud-hypervisor/
+
+---
+
+### POSIX & Application Support
+
+| | |
+|-|-|
+| **Difficulty** | 3/5 |
+| **Project Size** | Variable (175 or 350 hours) |
+| **Maximum instances** | 1 |
+| **Constraints/requirements** | Basic OS concepts, familiarity with POSIX and system calls, build systems and tool stacks. |
+
+#### Description
+
+One of the weak points of most unikernel projects has always been weak application support, often requiring that applications be ported to the target unikernel OS.
+With Unikraft we have been making a strong push towards POSIX compatibility so that applications can run unmodified on Unikraft.
+We have been doing this in two different ways: (1) by adding support to the musl libc library such that applications can be compiled against it, using their native build systems, and then linked into Unikraft and (2) through binary compatibility mode, where unmodified ELFs are directly executed in Unikraft and the resulting syscalls trapped and redirected to Unikraft code.
+
+This project may consist of, but is not limited to, any of the following items: improving Unikraft’s application support by (1) looking at the state of musl integration and analyzing to what extent unmodified application can be successfully linked and run with Unikraft; (2) conducting an equivalent analysis for binary compatibility mode and/or (3) implementation or extension of syscalls.
+The success of this project would provide an enormous boost, we think, to Unikraft adoption. The project length can be varied depending on which of these items are covered by the project.
+
+#### Reading & Related Material
+
+ * https://www.musl-libc.org/
+ * https://unikraft.org/docs/features/posix-compatibility/
+
+---
+
+### Unikraft & DevOps
+
+| | |
+|-|-|
+| **Difficulty** | 3/5 |
+| **Project Size** | Variable (175 or 350 hours) |
+| **Maximum instances** | 1 |
+| **Constraints/requirements** | Strong Linux command-line and tooling experience, Golang, Makefile & C programming skills. |
+
+#### Description
+
+With Unikraft, the core pillars of its development are found from its inherent performance, security benefits as well as its integrity.
+These properties can be facilitated with CI/CD systems which are designed for modern systems and provide the ability to run checks, tests, and audits for systems before they are deployed.
+Thanks to its modular architecture and multi-architecture and multi-platform capabilities, running tests on Unikraft results in many permutations of the same application – resulting in a large and complex CI/CD system.
```

Comment:
```suggestion
Thanks to its modular architecture and multi-platform capabilities, running tests on Unikraft results in many permutations of the same application – resulting in a large and complex CI/CD system.
```

"modular architecture and multi-architecture" mean the same thing, remove one of them